cmd/commands.go

```diff
@@ -374,16 +377,32 @@ func tryPropagateStatusesInventoryForDb(db store.DataStore, c cinv.Client, dbnam
 	}
 
 	var err error
+	var errReturned error
 	for _, status := range []string{"accepted", "pending", "rejected", "preauthorized"} {
 		err = updateDevicesStatus(ctx, db, c, tenant, status, dryRun)
 		if err != nil {
 			l.Infof("Done with DB %s status=%s, but there were errors: %s.", dbname, status, err.Error())
+			errReturned = err
 		} else {
 			l.Infof("Done with DB %s status=%s", dbname, status)
 		}
 	}
+	if migrationVersion != "" && !dryRun {
+		if err != nil {
```

Comment:
if the last updateDeviceStatus returns nil as error, you'll miss this if.
Are you sure here you want to check err instead of errReturned?

Reply:
good catch! thanks! fixed